content/docs/guides/remix.md

```diff
@@ -0,0 +1,189 @@
+
+
+To create a Neon project and access it from a Remix application:
+
+1. [Create a Neon project](#create-a-neon-project)
+2. [Create a Remix project and add dependencies](#create-a-remix-project-and-add-dependencies)
+3. [Configure a Postgres client](#configure-the-postgres-client)
+4. [Run the app](#run-the-app)
+
+## Create a Neon project
+
+If you do not have one already, create a Neon project. Save your connection details including your password. They are required when defining connection settings.
+
+1. Navigate to the [Projects](https://console.neon.tech/app/projects) page in the Neon Console.
+2. Click **New Project**.
+3. Specify your project settings and click **Create Project**.
+
+## Create a Remix project and add dependencies
+
+1. Create a Remix project if you do not have one. For instructions, see [Quick Start](https://remix.run/docs/en/main/start/quickstart), in the Remix documentation.
+
+2. Add project dependencies using one of the following commands:
+
+    <CodeTabs labels={["node-postgres", "postgres.js", "Neon serverless driver"]}>
+
+      ```shell
+      npm install pg
+      ```
+
+      ```shell
+      npm install postgres
+      ```
+
+      ```shell
+      npm install @neondatabase/serverless
+      ```
+
+    </CodeTabs>
+
+## Store your Neon credentials
+
+Add a `.env` file to your project directory and add your Neon connection string to it. You can find the connection string for your database in the **Connection Details** widget on the Neon **Dashboard**. For more information, see [Connect from any application](/docs/connect/connect-from-any-app).
+
+<CodeBlock shouldWrap>
+
+```shell
+DATABASE_URL=postgres://[user]:[password]@[neon_hostname]/[dbname]
+```
+
+</CodeBlock>
+
+## Configure the Postgres client
+
+TODO. Add explanation for why a `.db.server` file is required.
+
+### db.server
+
+Create a `db.server.ts` file at the root of your `/app` directory add the following code snippet to connect to your Neon database:
+
+
+<CodeTabs labels={["node-postgres", "postgres.js", "Neon serverless driver"]}>
+
+```javascript
+import pg from 'pg';
+
+const pool = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: true,
+  },
+});
+
+export { pool };
+```
+
+```javascript
+import postgres from 'postgres';
+
+const sql = postgres(process.env.DATABASE_URL, { ssl: 'require' });
+
+export { sql };
+```
+
+```javascript
+import { neon } from '@neondatabase/serverless';
+
+const sql = neon(process.env.DATABASE_URL);
+
+export { sql };
+```
+</CodeTabs>
+
+### route
+
+Create a new route in your `app/routes` directory and import the `db.server` file.  
+
+<CodeTabs labels={["node-postgres", "postgres.js", "Neon serverless driver"]}>
+
+```javascript
+import { json } from '@remix-run/node';
+import { useLoaderData } from '@remix-run/react';
+import { pool } from '~/db.server';
+
+export const loader = async () => {
+  const client = await pool.connect();
```

Comment:
This is probably just my ignorance, but is there a reason you want to start the connection here vs in the `db.server.ts` file?

Reply:
That's a good point. This guide is the first one we've done where the database / connection are abstracted away. All our other guides show the import and connect in the one file. 

In this case I think I might make sense to move the connection to `db.server.ts`. Great suggestion!

Reply:
Ok, i've looked into this. I don't think it makes anything any clearer. I will need to check with some folks to see if there are any benefits to abstracting the pool.connect. This is what I ended up with.




```
import pg from 'pg';

const getDb = async () => {
  const pool = new pg.Pool({
    connectionString: process.env.DATABASE_URL,
    ssl: true,
  });

  const client = await pool.connect();

  return client;
};

export { getDb };
```

```
import { json } from '@remix-run/node';
import { useLoaderData } from '@remix-run/react';
import { getDb } from '~/db.server';

export const loader = async () => {
  const client = await getDb();

  try {
    const response = await client.query('SELECT version()');
    console.log(response.rows[0]);
    return json({ data: response.rows[0] });
  } finally {
    client.release();
  }
};

export default function Page() {
  const data = useLoaderData();
  return <pre>{JSON.stringify(data, null, 2)}</pre>;
}
```